internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -116,6 +116,36 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 		}
 	}
 
+	// Set Taint to a node in an old machineDeployment and unset Taint from a node in a new machineDeployment
+	_, machineHasAnnotation := machine.Annotations[clusterv1.OldDeploymentAnnotation]
+	taint := corev1.Taint{Key: clusterv1.OldDeploymentAnnotation, Effect: corev1.TaintEffectPreferNoSchedule}
+	idx := -1
+	nodeHasTaint := false
+	for i, t := range node.Spec.Taints {
+		if t == taint {
+			idx = i
+			nodeHasTaint = true
+			break
+		}
+	}
```

Comment:
```suggestion
	idx := -1
	nodeHasTaint := false
	for i, t := range node.Spec.Taints {
		if t == taint {
			idx = i
			nodeHasTaint = true
			break
		}
	}
```
what about making this/the code below more readable by adding a new util/taint package with HasTaint, EnsureTaint, RemoveTaint (similarly to what we have for labels, annotations, ownerReferences even if not all of them are well orgainzed in separated packages)